modules/distributor/forwarder.go

```diff
@@ -77,25 +78,25 @@ func newForwarder(fn forwardFunc, o *overrides.Overrides) *forwarder {
 }
 
 // SendTraces queues up traces to be sent to the metrics-generators
-func (f *forwarder) SendTraces(ctx context.Context, tenantID string, keys []uint32, traces []*rebatchedTrace) {
+func (f *generatorForwarder) SendTraces(ctx context.Context, tenantID string, keys []uint32, traces []*rebatchedTrace) {
 	select {
 	case <-f.shutdown:
 		return
 	default:
 	}
 
-	qm := f.getOrCreateQueueManager(tenantID)
-	err := qm.pushToQueue(ctx, &request{keys: keys, traces: traces})
+	q := f.getOrCreateQueue(tenantID)
+	err := q.Push(ctx, &request{tenantID: tenantID, keys: keys, traces: traces})
```

Comment:
Nit: I don't think the tenantID is really needed. The queue already knows its tenant.

Reply:
Yeah, but perhaps not every queue user is interested in having the `tenantID` injected. I'd just keep it abstract. Also, if I were to remove it now, I'd have to either refactor the queue.ProcessFunc so that it accepts `tenantID` as a parameter or inject & extract it from context, which will only bloat the code IMO.